### PR TITLE
[maintenance] Fix enabled rules e2e test

### PR DESCRIPTION
**Pull request by tgodzik**
Previously there were not running on CI so it was possible they got broken in the meantime.

### CODE REVIEW

**Comment by tgodzik** [comment]
Any chance to get the tests enabled so that we can test if my fix works?

**Comment by xuansontrinh** [comment]
@hb-man Could you help set up the task "bazel test //e2e:enabled_rules_test" to test run on this PR only on TC?

**Comment by xuansontrinh** [comment]
@tgodzik could you try to check it out?

**Comment by tgodzik** [comment]
> @tgodzik could you try to check it out?

Yep, thanks!

**Comment by tgodzik** [comment]
Hmm.... any chance to get the actual log of the test? Do we need to change bazel for that? 

The logs only show `/home/teamcity/.cache/bazel/_bazel_teamcity/dda07a9d882174685c366a31fa4d36ed/execroot/_main/bazel-out/k8-fastbuild/testlogs/e2e/enabled_rules_test_bazel_6_4_0/test.log` :/

**Comment by hb-man** [comment]
@tgodzik hey!
are logs on build agent for test run + logs generated at report not enough?
https://bazel.teamcity.com/buildConfiguration/Bazel_TestBuilds_E2eTestsE2eEnabledRulesTest/10130?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandPull+Request+Details=true&expandBuildChangesSection=true

**Comment by hb-man** [comment]
> Hmm.... any chance to get the actual log of the test? Do we need to change bazel for that?
> 
> The logs only show `/home/teamcity/.cache/bazel/_bazel_teamcity/dda07a9d882174685c366a31fa4d36ed/execroot/_main/bazel-out/k8-fastbuild/testlogs/e2e/enabled_rules_test_bazel_6_4_0/test.log` :/

okay, sorry, i think i misunderstood your question:
if you open the[ test page on TC](https://bazel.teamcity.com/buildConfiguration/Bazel_TestBuilds_E2eTestsE2eEnabledRulesTest?mode=builds), you'll find there a test report you can download them
image


Here are the direct links to logs:
[6.4.0](https://bazel.teamcity.com/repository/download/Bazel_TestBuilds_E2eTestsE2eEnabledRulesTest/10130:id/testlogs.zip!/5a0f72b96e927ac8d65856e77ad031b7/execroot/_main/bazel-out/k8-fastbuild/testlogs/e2e/enabled_rules_test_bazel_6_4_0/test.log)
[5.3.2](https://bazel.teamcity.com/repository/download/Bazel_TestBuilds_E2eTestsE2eEnabledRulesTest/10130:id/testlogs.zip!/5a0f72b96e927ac8d65856e77ad031b7/execroot/_main/bazel-out/k8-fastbuild/testlogs/e2e/enabled_rules_test_bazel_5_3_2/test.log)

**Comment by tgodzik** [comment]
I think this is mostly fixed, but we need https://github.com/build-server-protocol/bsp-testkit2/pull/46

**Comment by hb-man** [comment]
@agluszak probably can help with review of https://github.com/build-server-protocol/bsp-testkit2/pull/46 ?

**Comment by tgodzik** [comment]
Could we try and run the added tests here? Any chance to have it run permanently or only after merge?

**Comment by hb-man** [comment]
Strange, they should've been run on each commit.
I've run them manually right now, let's wait for results, and i'll fix so they run on each commit too in case test is failed

**Comment by hb-man** [comment]
Okay, tests are still failing. I've fixed the problem with automatic run, this test should be run for any commit landing in this PR.

**Comment by tgodzik** [comment]
I am a bit out of ideas. I can run the test locally and it works but on CI on ScalaTarget data is being set at all. No JVM either, any idea what might be the difference here?

**Comment by tgodzik** [comment]
Ok, I figured finally what was happening. projectview was being overriden and that's why the test was failing. Should be good now

**Comment by xuansontrinh** [comment]
The PR looks good to me.
@hb-man could you please help me double-check the TC config?

**Comment by xuansontrinh** [comment]
The PR looks good to me. Just a few optional suggestions left.
@hb-man could you please help me double-check the TC config?

**Comment by hb-man** [comment]
Yep, config is correct and once the PR in master it will be added to our pipeline.

**Comment by tgodzik** [comment]
Should we merge it or there is something more to fix?

**Comment by xuansontrinh** [comment]
@tgodzik I left some comments for you a few days ago.

**Comment by tgodzik** [comment]
> @tgodzik I left some comments for you a few days ago. It's minor though.

I must be blind, I don't see any. Could you point me to those?

**Review by xuansontrinh — COMMENTED** [review]
Here are the minor changes that I mentioned.
Please let me know if you see them now.

File: e2e/src/main/kotlin/org/jetbrains/bsp/bazel/BazelBspScalaProjectTest.kt [hunk 1]
```diff
@@ -1,55 +1,123 @@
...
       )
     }
     val expectedScalaOptionsResult = ScalacOptionsResult(expectedScalaOptionsItems)
     val scalaOptionsParams = ScalacOptionsParams(expectedTargetIdentifiers)
-    return BazelBspTestScenarioStep("scalaOptions results") {
-      testClient.testScalacOptions(60.seconds, scalaOptionsParams, expectedScalaOptionsResult)
+    val step = BazelBspTestScenarioStep("scalaOptions results") {
+      testClient.testScalacOptions(120.seconds, scalaOptionsParams, expectedScalaOptionsResult)
     }
+    return step
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply e2e/src/main/kotlin/org/jetbrains/bsp/bazel/BazelBspScalaProjectTest.kt]: leftover from debugging, thanks!

File: e2e/src/main/kotlin/org/jetbrains/bsp/bazel/base/BazelBspTestBaseScenario.kt [hunk 2]
```diff
@@ -21,6 +21,8 @@ abstract class BazelBspTestBaseScenario {
 
   val targetPrefix = calculateTargetPrefix()
 
+  open fun additionalOptions(): Array<String> = emptyArray()
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply e2e/src/main/kotlin/org/jetbrains/bsp/bazel/base/BazelBspTestBaseScenario.kt]: Changed

**Comment by tgodzik** [comment]
> Here are the minor changes that I mentioned. Please let me know if you see them now.

Yes, and should be fixed. Thanks!

**Review by xuansontrinh — APPROVED** [review]

**Pull request merged**